#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_use_sox_12.dpatch by  <alban.peignier@free.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Use /usr/bin/sox-12 instead of default sox binary

@DPATCH@

diff --git a/scripts/rd_export_file b/scripts/rd_export_file
index 4cc1823..b4ac3a9 100755
--- a/scripts/rd_export_file
+++ b/scripts/rd_export_file
@@ -78,6 +78,8 @@ PEAK=${11}
 WORK=${12}
 CUSTOM_CMD=${13}
 
+SOX=/usr/bin/sox-12
+
 #
 # Resampling Flag
 #
@@ -125,7 +127,7 @@ fi
 # Get Peak Level
 #
 if [ $NORMAL_LEVEL != 0 ] ; then
-  PEAK_LEVEL=`sox $WORK -e stat -v 2>&1 | grep -v ^sox` 
+  PEAK_LEVEL=`$SOX $WORK -e stat -v 2>&1 | grep -v ^sox` 
   SCALE=`echo "$NORMAL_LEVEL * $PEAK_LEVEL" | bc -l`
   if [ "$SCALE" == "1.00000000000000000000" ]; then
     SCALE="1"
@@ -136,7 +138,7 @@ fi
 
 case "$FORMAT_OUT" in
   0)  # PCM16
-    sox $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT "$FILE_OUT" $RESAMPLE_FLAG
+    $SOX $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT "$FILE_OUT" $RESAMPLE_FLAG
   ;;
 
   1)  # MPEG Layer 1
@@ -144,19 +146,19 @@ case "$FORMAT_OUT" in
   ;;
 
   2)  # Mpeg Layer 2
-    sox $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t wav - $RESAMPLE_FLAG | toolame -t 0 -s $TOOLAME_SAMPRATE -m $MPEG_MODE -b $BITRATE_OUT /dev/stdin "$FILE_OUT"
+    $SOX $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t wav - $RESAMPLE_FLAG | toolame -t 0 -s $TOOLAME_SAMPRATE -m $MPEG_MODE -b $BITRATE_OUT /dev/stdin "$FILE_OUT"
   ;;
 
   3)  # MPEG Layer 3
-    sox $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $RESAMPLE_FLAG | lame --silent -r -x -s $SAMPRATE_OUT -m $MPEG_MODE $LAME_RATE - "$FILE_OUT"
+    $SOX $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $RESAMPLE_FLAG | lame --silent -r -x -s $SAMPRATE_OUT -m $MPEG_MODE $LAME_RATE - "$FILE_OUT"
   ;;
 
   4)  # FLAC
-    sox $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $RESAMPLE_FLAG | flac -f --silent --best --endian=little --sign=signed --bps=16 --channels=$CHANS_OUT --sample-rate=$SAMPRATE_OUT --force-raw-format -o "$FILE_OUT" -
+    $SOX $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $RESAMPLE_FLAG | flac -f --silent --best --endian=little --sign=signed --bps=16 --channels=$CHANS_OUT --sample-rate=$SAMPRATE_OUT --force-raw-format -o "$FILE_OUT" -
   ;;
 
   5)  # OggVorbis
-    sox $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $RESAMPLE_FLAG | oggenc -r -B 16 -C $CHANS_OUT -R $SAMPRATE_OUT -q $QUALITY_OUT -o "$FILE_OUT" -
+    $SOX $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $RESAMPLE_FLAG | oggenc -r -B 16 -C $CHANS_OUT -R $SAMPRATE_OUT -q $QUALITY_OUT -o "$FILE_OUT" -
   ;;
 
   *)  # Custom Format
@@ -165,7 +167,7 @@ case "$FORMAT_OUT" in
       rm -f $PEAK
       exit 1
     fi
-    sox $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $RESAMPLE_FLAG | $CUSTOM_CMD
+    $SOX $SOX_SCALE $WORK -r $SAMPRATE_OUT -c $CHANS_OUT -t raw - $RESAMPLE_FLAG | $CUSTOM_CMD
   ;;
 esac
 
diff --git a/scripts/rd_import_file b/scripts/rd_import_file
index ff3fb72..deffb38 100755
--- a/scripts/rd_import_file
+++ b/scripts/rd_import_file
@@ -70,6 +70,8 @@ FILE_OUT=${9}
 PEAK=${10}
 WORK=${11}
 
+SOX=/usr/bin/sox-12
+
 set -e
 
 EXTENSION_IN=`echo $FILE_IN | sed 's/.*\.\([a-zA-Z0-9]\)/\1/'`
@@ -175,7 +177,7 @@ case "$FORMAT_OUT" in
 	if [ $3 != 1  ]; then
 	    SOX_OPTIONS="-v $3 $SOX_OPTIONS"
     fi
-    sox $SOX_OPTIONS | rdfilewrite --channels=$4 --sample-rate=$5 $2
+    $SOX $SOX_OPTIONS | rdfilewrite --channels=$4 --sample-rate=$5 $2
     }
     ;;
     1)
@@ -208,7 +210,7 @@ case "$FORMAT_OUT" in
             ;;
         esac
         LAME_OPTIONS="-t 0 -m $MODE -s $LAMERATE -b $BITRATE_OUT"
-	sox $SOX_OPTIONS | toolame $LAME_OPTIONS /dev/stdin -W $2 > /dev/null 2> /dev/null
+	$SOX $SOX_OPTIONS | toolame $LAME_OPTIONS /dev/stdin -W $2 > /dev/null 2> /dev/null
     }
     ;;
     *)
@@ -222,7 +224,7 @@ convert "$FILE_IN" $WORK
 # TODO: extract the PEAK_LEVEL during the convertion
 SCALE=1
 if [ $NORMAL_LEVEL != 0 ]; then
-    PEAK_LEVEL=`sox $WORK -t wav /dev/null stat -v 2>&1 | grep -v ^sox` 
+    PEAK_LEVEL=`$SOX $WORK -t wav /dev/null stat -v 2>&1 | grep -v ^sox` 
     # grep avoid sox warning messages
     
     SCALE=`echo "$NORMAL_LEVEL * $PEAK_LEVEL" | bc -l`
